internal/resources/kubernetessecret/scope/scope.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package scope
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"golang.org/x/exp/slices"
+
+	secretclustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster"
+)
+
+var (
+	ScopeSchema = &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Scope for the secret having one of the valid scopes for secret: cluster or cluster_group",
+		Required:    true,
+		ForceNew:    true,
+		MaxItems:    1,
+		MinItems:    1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				ClusterKey: ClusterPolicyFullname,
```

Comment:
Should there be cluster group scope also, as per the description of the schema you have mentioned?

Reply:
changed the description